docs: Explicitly mention ODBC driver setup (#349)
spiceai/docs: spiceaidocs/docs/components/data-connectors/odbc.md

description: 'ODBC Data Connector Documentation'
---

## Setup

:::warning

ODBC support is not included in the released binaries. To use ODBC with Spice, you need to [checkout and compile the code](https://github.com/spiceai/spiceai/blob/trunk/CONTRIBUTING.md#building) with the `--features odbc` flag (`cargo build --release --features odbc`).

:::

> An ODBC connection requires a compatible ODBC driver and valid driver configuration. ODBC drivers are available from their respective vendors. Here are a few examples:

Reviewer: It requires two components, a driver manager (i.e. `unixodbc`) and a driver to connect to the database. On Windows the driver manager is built-in, but Linux and macOS require `unixodbc` to be installed

description: 'ODBC Data Connector Documentation'
---

## Setup

:::warning

ODBC support is not included in the released binaries. To use ODBC with Spice, you need to [checkout and compile the code](https://github.com/spiceai/spiceai/blob/trunk/CONTRIBUTING.md#building) with the `--features odbc` flag (`cargo build --release --features odbc`).

:::

> An ODBC connection requires a compatible ODBC driver. Whether running as a Docker image or directly on a computer, a valid ODBC driver configuration is necessary. ODBC drivers for various connection types are available from their respective vendors:

Reviewer 2: ```suggestion
An ODBC connection requires a compatible ODBC driver. When run as a Docker image or on a computer, valid ODBC driver configuration is necessary. There are many ODBC drivers available from their respective vendors. For example:
```

> An ODBC connection requires a compatible ODBC driver. Whether running as a Docker image or directly on a computer, a valid ODBC driver configuration is necessary. ODBC drivers for various connection types are available from their respective vendors:

Reviewer 2: ```suggestion
An ODBC connection requires a compatible ODBC driver and valid driver configuration. ODBC drivers are available from their respective vendors.  Here are a few examples:
```

- [PostgreSQL](https://odbc.postgresql.org/)
- [MySQL](https://dev.mysql.com/downloads/connector/odbc/)
- [Databricks](https://www.databricks.com/spark/odbc-drivers-download)
- [AWS Athena](https://docs.aws.amazon.com/athena/latest/ug/connect-with-odbc.html)

> Please note that this is not an exhaustive list of drivers, as there are many ODBC adapters available.

Reviewer 2: I think you can remove this sentence with the above suggestion.

description: 'ODBC Data Connector Documentation'
---

## Setup

:::warning

> ODBC support is not included in the released binaries. To use ODBC with Spice, you need to [checkout and compile the code](https://github.com/spiceai/spiceai/blob/trunk/CONTRIBUTING.md#building) with the `--features odbc` flag (`cargo build --release --features odbc`).

Reviewer 3: Or use the official Docker image - see language from release notes: https://github.com/spiceai/spiceai/pull/2132/files

- [Databricks](https://www.databricks.com/spark/odbc-drivers-download)
- [AWS Athena](https://docs.aws.amazon.com/athena/latest/ug/connect-with-odbc.html)

> Unix additionally requires the installation of `unixodbc`. Here are some examples for installing this package:

Reviewer: ```suggestion
Non-Windows systems additionally require the installation of an ODBC Driver Manager like `unixodbc`.
```

Reviewer 3: "In addition, ..."
